cordformation/src/main/kotlin/net/corda/plugins/Cordapp.kt

```diff
@@ -1,15 +1,28 @@
 package net.corda.plugins
 
 import org.gradle.api.Project
+import org.gradle.api.tasks.Input
+import org.gradle.api.tasks.Optional
 import java.io.File
 
-open class Cordapp private constructor(val coordinates: String?, val project: Project?) {
+open class Cordapp private constructor(
+    @get:[Optional Input] val coordinates: String?,
+    @get:[Optional Input] val project: Project?
+) {
     constructor(coordinates: String) : this(coordinates, null)
     constructor(cordappProject: Project) : this(null, cordappProject)
 
     // The configuration text that will be written
+    @Optional
+    @Input
     internal var config: String? = null
 
+    /**
+     * Determines whether or not CordFormation with deploy this CorDapp into Corda.
+     */
+    @Input
+    var deployed: Boolean = true
```

Comment:
In terms of naming this makes more sense from a DSL perspective, which are declarative, to be "deploy" or "enable" as opposed to "deployed" or "enabled".

Reply:
I had to changed from `enabled` because it was ambiguous with an existing `enabled` field - probably the task's.